cugraph_nx_demo.ipynb

```diff
@@ -0,0 +1,504 @@
+    }
+   ],
+   "source": [
+    "nx.__version__, pd.__version__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The 'email-Eu-core' dataset, made up on anonymized email data from a large European institution, is a directed graph with 25571 edges and 1005 nodes. The following block reads this dataset and loads into a networkx.Graph object. We then measure the performance of the betweenness centrality algorithm with an approximation of k=50 node samples."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_csv(\"datasets/email-Eu-core.csv\", header=None, names=['src', 'dst', 'wgt'], sep=' ')\n",
+    "Gnx = nx.from_pandas_edgelist(df, source='src', target='dst', create_using=nx.Graph())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "553 ms ± 7.86 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "bc = nx.betweenness_centrality(Gnx, k=50)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Customization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Depending on your graph needs, `networkx` supports the ability to choose your dispatcher. A dispatchable networkX algorithm will use the chosen dispatcher to convert networkx Graph to the backend equivalent, dispatch the work, then convert back to the required nx form. More information here: https://networkx.org/documentation/stable/reference/classes/index.html#backends"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cugraph_nx as cnx"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "('3.2rc0.dev0', '2.0.3', '23.10.00')"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "nx.__version__, pd.__version__, cnx.__version__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Method 1: Specify which backend to use within function call"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "141 ms ± 2.77 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "bc = nx.betweenness_centrality(Gnx, k=50, backend='cugraph')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Method 2: Perform backend conversion on a separate line"
```

Comment:
I'm thinking method 2 should be less common, but we're actually going over these use cases so there's a good chance these may change.  Regardless of that, this needs to explain why someone would use one method over the other, the tradeoffs of one over the other, etc.

Reply:
I think this will be rewritten substantially, since the API is going many changes and quite a bit isn't set in stone. However, just looking at these two methods, I agree that method 1 is easier to use over method 2. Method 1 is conducive to devs refactoring their code to introduce using one of the backends, but even then setting a dispatching order in a configs file / location makes more sense. I suppose Method 2 is helpful for explicitly showing the conversion and also if the same Graph object is going to be used across multiple calls. I'll make a thread here as I compare the methods/use cases more intensely.